# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1768156651 -3600
# Parent  34155f3e74bab49d3dd749c97ed3446d753328d4
Bug 2009620 - Don't expand empty recent/favourites submenu.

This is not perfect since the menu entry to open the submenu should be disabled earlier, but better than nothing.
Besides: TB's new motto is: "Perfection is the enemy of good".

diff --git a/mailnews/base/content/folder-menupopup.js b/mailnews/base/content/folder-menupopup.js
--- a/mailnews/base/content/folder-menupopup.js
+++ b/mailnews/base/content/folder-menupopup.js
@@ -657,8 +657,9 @@
         });
 
         if (specialFoldersMap.length == 0) {
           menu.setAttribute("disabled", "true");
+          submenu.setAttribute("hidden", "true");
         }
 
         this._initializedSpecials.add(specialType);
       }
